Source/UI/CommonUI/View/Component/BorderView.swift

```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+final class BorderView: UIView {
+
+    private var style: ElementBorderStyle?
+
+    /// shaper layer that draw edges and corners
+    private var borderLayer = CAShapeLayer()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        layer.addSublayer(borderLayer)
+    }
+
+    // Keep border layer size in sync with owning view
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        borderLayer.frame = bounds
+        guard let style = style else { return }
+        borderLayer.createCustomBorder(with: style)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    /// Update edges and corners style
+    /// - Parameter style: protocol with some parameters for styling border and corners
+    func update(with style: ElementBorderStyle) {
+        self.style = style
```

Comment:
I know its not currently supported, but as it stands if a style is overridden it will never materialise. We could fix by:
```suggestion
        self.style = style
        borderLayer.removeFromSuperlayer()
        borderLayer = CAShapeLayer()
        borderLayer.createCustomBorder(with: style)
        layer.addSublayer(borderLayer)
```

And in fairness once we do this, we no longer need to create the border layer on layout subviews, we create it when a style is provided. And we can now have the source of truth on whether to replace the layout in the style provided being different from last or not.
```swift
guard style != self.style else { return }
```